datajunction-server/datajunction_server/construction/build_v2.py

```diff
@@ -0,0 +1,937 @@
+            else:
+                measure_columns.append(expr)
+                expr.set_semantic_type(SemanticType.MEASURE)  # type: ignore
+        await parent_ast.compile(context)
+
+        # Build translated SQL object
+        columns_metadata = [
+            assemble_column_metadata(  # pragma: no cover
+                cast(ast.Column, col),
+            )
+            for col in parent_ast.select.projection
+        ]
+        dependencies, _ = await parent_ast.extract_dependencies(
+            CompileContext(session, DJException()),
+        )
+        measures_sql_mapping[parent_node.name] = TranslatedSQL(
+            sql=str(parent_ast),
+            columns=columns_metadata,
+            dialect=build_criteria.dialect,
+            upstream_tables=[
+                f"{dep.catalog.name}.{dep.schema_}.{dep.table}"
+                for dep in dependencies
+                if dep.type == NodeType.SOURCE
+            ],
+        )
+    return measures_sql_mapping
+
+
+async def build_node(  # pylint: disable=too-many-arguments,too-many-locals
+    session: AsyncSession,
+    node: NodeRevision,
+    filters: Optional[List[str]] = None,
+    dimensions: Optional[List[str]] = None,
+    orderby: Optional[List[str]] = None,  # pylint: disable=unused-argument
+    limit: Optional[int] = None,  # pylint: disable=unused-argument
+    build_criteria: Optional[BuildCriteria] = None,
+    access_control: Optional[access.AccessControlStore] = None,
+    include_dimensions_in_groupby: bool = None,
+) -> ast.Query:
+    """
+    Builds the node SQL with the requested set of dimensions, filter expressions,
+    order by, and limit clauses.
+
+    Build Strategy
+    ---------------
+    1. Recursively turn node references into query ASTs + apply any filters that can
+    be pushed down. If the node query has CTEs, unwind them into the query.
+    2. Initialize the final query with the node's query AST added to it as a CTE.
+    3. For any dimensions or filters requested for the node, determine if a join is
+    needed to bring in the dimension or filter. Keep track of all the necessary dimension
+    joins in a dict that maps dimension nodes to join paths.
+    4. For each of the necessary dimension joins, build the dimension node's query in the
+    same manner as above, recursively replacing any node references and pushing down requested
+    filters where possible.
+    5. Add each dimension node's query AST to the final query as a CTE.
+    6. Build the final query using the various CTEs. This does all the joins between the node
+    query AST and the dimension nodes' ASTs using the join logic from the dimension links.
+    7. Add all requested dimensions to the final select.
+    8. Add order by and limit to the final select (TODO)
+    """
+    if access_control:
+        access_control.add_request_by_node(node)
+
+    # Set the dialect by finding available engines for this node, or default to Spark
```

Comment:
Comment does not seem to match the code.

Reply:
Ahh yes, let me remove that!